esp-hal/src/interrupt/mod.rs

```diff
@@ -125,6 +125,50 @@ pub trait InterruptConfigurable: crate::private::Sealed {
     fn set_interrupt_handler(&mut self, handler: InterruptHandler);
 }
 
+/// Represents an ISR callback function
+#[derive(Copy, Clone, Debug)]
+#[cfg_attr(feature = "defmt", derive(defmt::Format))]
+pub struct IsrCallback {
+    f: usize,
```

Comment:
This should be NonZeroUsize otherwise `Option<IsrCallback>` will not be guaranteed to be 4 bytes